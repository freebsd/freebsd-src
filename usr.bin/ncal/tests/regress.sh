
CAL_BIN="ncal"
CAL="${CAL_BIN} -C"
NCAL="${CAL_BIN} -N"
YEARS="2008 2009 2010 2011"
ONEYEAR="2009"

echo 1..89

REGRESSION_START($1)

#
# The first tests are layout tests, to make sure that the output is still the
# same despite varying months.
#

# Full year calendars

for y in ${YEARS}; do
	# Regular calendar, Month days, No-highlight
	REGRESSION_TEST(`r-y${y}-md-nhl', `$NCAL -h ${y}')
	# Backwards calendar, Month days, No-highlight
	REGRESSION_TEST(`b-y${y}-md-nhl', `$CAL -h ${y}')
	# Regular calendar, Julian days, No-highlight
	REGRESSION_TEST(`r-y${y}-jd-nhl', `$NCAL -jh ${y}')
	# Backwards calendar, Julian days, No-highlight
	REGRESSION_TEST(`b-y${y}-jd-nhl', `$CAL -jh ${y}')
done

# 3 month calendars

for m in $(jot -w %02d 12); do
	# Regular calendar, Month days, No-highlight
	REGRESSION_TEST(`r-3m${ONEYEAR}${m}-md-nhl',
	    `$NCAL -h3 ${m} ${ONEYEAR}')
	# Backwards calendar, Month days, No-highlight
	REGRESSION_TEST(`b-3m${ONEYEAR}${m}-md-nhl', `$CAL -h3 ${m} ${ONEYEAR}')
	# Regular calendar, Julian days, No-highlight
	REGRESSION_TEST(`r-3m${ONEYEAR}${m}-jd-nhl',
	    `$NCAL -jh3 ${m} ${ONEYEAR}')
	# Backwards calendar, Julian days, No-highlight
	REGRESSION_TEST(`b-3m${ONEYEAR}${m}-jd-nhl', `$CAL -jh3 ${m} ${ONEYEAR}')
done

#
# The next tests are combinations of the various arguments.
#

# These should fail
REGRESSION_TEST(`f-3y-nhl',  `$NCAL -3 -y 2>&1')
REGRESSION_TEST(`f-3A-nhl',  `$NCAL -3 -A 3 2>&1')
REGRESSION_TEST(`f-3B-nhl',  `$NCAL -3 -B 3 2>&1')
REGRESSION_TEST(`f-3gy-nhl', `$NCAL -3 2008 2>&1')
REGRESSION_TEST(`f-3AB-nhl', `$NCAL -3 -A 3 -B 3 2>&1')
REGRESSION_TEST(`f-mgm-nhl', `$NCAL -m 3 2 2008 2>&1')
REGRESSION_TEST(`f-ym-nhl',  `$NCAL -y -m 2 2>&1')
REGRESSION_TEST(`f-ygm-nhl', `$NCAL -y 2 2008 2>&1')
REGRESSION_TEST(`f-yA-nhl',  `$NCAL -y -A 3 2>&1')
REGRESSION_TEST(`f-yB-nhl',  `$NCAL -y -B 3 2>&1')
REGRESSION_TEST(`f-yAB-nhl', `$NCAL -y -A 3 -B 3 2>&1')

# These should be successful

REGRESSION_TEST(`s-b-3-nhl',    `$CAL -d 2008.03 -3')
REGRESSION_TEST(`s-b-A-nhl',    `$CAL -d 2008.03 -A 1')
REGRESSION_TEST(`s-b-B-nhl',    `$CAL -d 2008.03 -B 1')
REGRESSION_TEST(`s-b-AB-nhl',   `$CAL -d 2008.03 -A 1 -B 1')
REGRESSION_TEST(`s-b-m-nhl',    `$CAL -d 2008.03 -m 1')
REGRESSION_TEST(`s-b-mgy-nhl',  `$CAL -d 2008.03 -m 1 2007')
REGRESSION_TEST(`s-b-gmgy-nhl', `$CAL -d 2008.03 1 2007')
REGRESSION_TEST(`s-r-3-nhl',    `$NCAL -d 2008.03 -3')
REGRESSION_TEST(`s-r-A-nhl',    `$NCAL -d 2008.03 -A 1')
REGRESSION_TEST(`s-r-B-nhl',    `$NCAL -d 2008.03 -B 1')
REGRESSION_TEST(`s-r-AB-nhl',   `$NCAL -d 2008.03 -A 1 -B 1')
REGRESSION_TEST(`s-r-m-nhl',    `$NCAL -d 2008.03 -m 1')
REGRESSION_TEST(`s-r-mgy-nhl',  `$NCAL -d 2008.03 -m 1 2007')
REGRESSION_TEST(`s-r-gmgy-nhl', `$NCAL -d 2008.03 1 2007')

REGRESSION_END()
