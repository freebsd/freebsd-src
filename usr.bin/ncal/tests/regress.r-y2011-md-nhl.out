                                  2011
    January           February          March             April             
Mo     3 10 17 24 31     7 14 21 28        7 14 21 28        4 11 18 25   
Tu     4 11 18 25     1  8 15 22        1  8 15 22 29        5 12 19 26   
We     5 12 19 26     2  9 16 23        2  9 16 23 30        6 13 20 27   
Th     6 13 20 27     3 10 17 24        3 10 17 24 31        7 14 21 28   
Fr     7 14 21 28     4 11 18 25        4 11 18 25        1  8 15 22 29   
Sa  1  8 15 22 29     5 12 19 26        5 12 19 26        2  9 16 23 30   
Su  2  9 16 23 30     6 13 20 27        6 13 20 27        3 10 17 24      

    May               June              July              August            
Mo     2  9 16 23 30     6 13 20 27        4 11 18 25     1  8 15 22 29   
Tu     3 10 17 24 31     7 14 21 28        5 12 19 26     2  9 16 23 30   
We     4 11 18 25     1  8 15 22 29        6 13 20 27     3 10 17 24 31   
Th     5 12 19 26     2  9 16 23 30        7 14 21 28     4 11 18 25      
Fr     6 13 20 27     3 10 17 24        1  8 15 22 29     5 12 19 26      
Sa     7 14 21 28     4 11 18 25        2  9 16 23 30     6 13 20 27      
Su  1  8 15 22 29     5 12 19 26        3 10 17 24 31     7 14 21 28      

    September         October           November          December          
Mo     5 12 19 26        3 10 17 24 31     7 14 21 28        5 12 19 26   
Tu     6 13 20 27        4 11 18 25     1  8 15 22 29        6 13 20 27   
We     7 14 21 28        5 12 19 26     2  9 16 23 30        7 14 21 28   
Th  1  8 15 22 29        6 13 20 27     3 10 17 24        1  8 15 22 29   
Fr  2  9 16 23 30        7 14 21 28     4 11 18 25        2  9 16 23 30   
Sa  3 10 17 24        1  8 15 22 29     5 12 19 26        3 10 17 24 31   
Su  4 11 18 25        2  9 16 23 30     6 13 20 27        4 11 18 25      
